standardtracer/propagation.go

```diff
@@ -177,52 +181,48 @@ func (p *splitBinaryPropagator) JoinTrace(
 	var traceID, propagatedSpanID int64
 	var sampledByte byte
 
-	err = binary.Read(contextReader, binary.BigEndian, &traceID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &traceID); err != nil {
 		return nil, opentracing.ErrTraceCorrupted
 	}
-	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &propagatedSpanID); err != nil {
```

Comment:
Very briefly, the idea with `standardtracer` was to try and provide a reasonable / generic in-memory representation of an active / extant Span while coupling as little as possible to a particular tracing data model. When we removed TraceContext/TraceContextSource from OT I caved and made `standardtracer` way less general. In so doing it became rather pointless from a practical perspective (if you ask me).

As @yurishkuro said, @tschottdorf is clearly doing great work here so it seems a shame to throw `standardtracer` out! One possibility would be to re-introduce the TraceContext-like abstractions and make it more general while keeping most of the performance improvements here. Another possibility would be to lay the groundwork for a lightweight, OpenTracing-native multilingual tracing *system* that integrates with all OT platforms and provides a basic but useful in-memory-only backend. A third possibility would be to leave it more or less as-is.